beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -16,14 +16,23 @@ import (
 	enginev1 "github.com/prysmaticlabs/prysm/proto/engine/v1"
 	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
 	"github.com/prysmaticlabs/prysm/runtime/version"
+	"github.com/sirupsen/logrus"
 )
 
-func (vs *Server) getBellatrixBeaconBlock(ctx context.Context, req *ethpb.BlockRequest) (*ethpb.BeaconBlockBellatrix, error) {
+func (vs *Server) getBellatrixBeaconBlock(ctx context.Context, req *ethpb.BlockRequest) (*ethpb.GenericBeaconBlock, error) {
 	altairBlk, err := vs.buildAltairBeaconBlock(ctx, req)
 	if err != nil {
 		return nil, err
 	}
 
+	builderReady, b, err := vs.getAndBuildHeaderBlock(ctx, altairBlk)
+	if err != nil {
+		// In the event of an error, the node should fall back to default execution engine for building block.
+		log.WithError(err).Error("Default back to local execution client")
+	} else if builderReady {
```

Comment:
would it be better to do 
```
if err != nil {
...
} else {
...
if builderReady {
...
}
}
```
just so no one can accidentally change that code to return builderReady = true + error

Reply:
nvm here